src/spec.ts

```diff
@@ -319,9 +319,31 @@ export function alwaysNoOcclusion(spec: ExtendedUnitSpec): boolean {
   return vlEncoding.isAggregate(spec.encoding);
 }
 
+function hasFieldDef(fieldDefs: FieldDef[], fieldDef: FieldDef): boolean {
+  for (let i = 0; i < fieldDefs.length; i++) {
+    if (identical(fieldDefs[i], fieldDef)) {
+      return true;
+    }
+  }
+  return false;
+}
+
 export function fieldDefs(spec: ExtendedUnitSpec | LayerSpec): FieldDef[] {
```

Comment:
Out of curiosity, is `FacetSpec` being used in the examples right now? I don't see "facet" in the example specs